### PR TITLE
chore(sql): fixed cast with index parse error

### PR DESCRIPTION
**Pull request by bziobrowski**
Fixes "cast(s as symbol index)" by removing index handling from this element so separate index(column_name) should be used.  
Cast with index doesn't create index in master anyway .

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|3|3|100.00%|

File: core/src/test/java/io/questdb/cairo/CreateTableTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        assertColumnsIndexed("new", "s");
+    }
+
+    private void assertColumnsIndexed(String tableName, String... columnNames) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader r = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                TableReaderMetadata metadata = r.getMetadata();
+                IntList indexed = new IntList();
+
+                for (int i = 0, cols = metadata.getColumnCount(); i < cols; i++) {
```
Comment (ideoma) [inline core/src/test/java/io/questdb/cairo/CreateTableTest.java:170]: minor: this for is equivalent to 1 line
```suggestion
                indexed.setPos(metadata.getColumnCount());
```

**Review by ideoma — APPROVED** [review]
That's first time I see `cast` after create table. We need some docs about it

**Comment by bziobrowski** [comment]
@ideoma It is sort-of documented at https://questdb.io/docs/reference/sql/create-table/ , e.g. 

```
CREATE TABLE x AS (SELECT * FROM table WHERE false),
  cast(price AS LONG),
  cast(instrument as SYMBOL INDEX);
```
 though there are no examples that show  that cast accepts almost the same syntax as column def within create table .

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|3|3|100.00%|

File: core/src/test/java/io/questdb/cairo/CreateTableTest.java (hunk 1 above)
Comment (bziobrowski) [reply core/src/test/java/io/questdb/cairo/CreateTableTest.java]: Fixed

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|3|3|100.00%|

**Pull request merged**